File: solutions/cpp/1522/main.cpp
#include<bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    string S; cin >> S;
    int N = (int)S.size();
    int cntA = count(S.begin(), S.end(), 'a');
    S += S;
    int ans = INT_MAX, cntB = 0;
    for(int i = 0, r = -1; i < N; ++i) {
        while(r + 1 < i + cntA) {
            if(S[++r] == 'b') ++cntB;
        }
        ans = min(ans, cntB);
        if(S[i] == 'b') --cntB;
    }
    cout << ans;

    return 0;
}

File: solutions/python/1522/main.py
import sys


def input() -> str:
    return sys.stdin.readline().rstrip()


def solution(S: str) -> int:
    N = len(S)
    cntA = S.count("a")
    S = S + S
    ans = N + 1
    cntB = 0
    right = -1
    for i in range(N):
        while right + 1 < i + cntA:
            right += 1
            if S[right] == "b":
                cntB += 1
        ans = min(ans, cntB)
        if S[i] == "b":
            cntB -= 1
    return ans


if __name__ == "__main__":
    S = input()
    answer = solution(S)
    print(answer)
